src/tests/test_magic.py

```diff
@@ -1989,6 +1989,71 @@ def test_accessing_previously_nonexisting_file(ip_empty, tmp_empty, capsys):
     assert expected in out
 
 
+@pytest.mark.parametrize(
+    "query, desired_error_msgs, unwanted_error_msgs",
+    [
+        (
+            "%sql select not_a_function(body_mass_g) from mysnippet",
+            [
+                "Scalar Function with name not_a_function does not exist!",
+            ],
+            [
+                "There is no table with name 'mysnippet'",
+                "Did you mean: 'mysnippet'",
+                "Table with name mysnippet does not exist!",
+            ],
+        ),
+        (
+            "%sql select not_a_function(body_mass_g) from mysnip",
+            [
+                "If using snippets, you may pass the --with argument explicitly.",
+                "There is no table with name 'mysnip'",
+                "Table with name mysnip does not exist!",
+            ],
+            [
+                "Scalar Function with name not_a_function does not exist!",
+            ],
+        ),
+    ],
+    ids=[
+        "no-snippet-typo",
+        "snippet-typo",
+    ],
+)
+def test_query_snippet_bad_function_error_message(
+    ip_empty, query, desired_error_msgs, unwanted_error_msgs
+):
+    # Set up snippet
+    ip_empty.run_cell("%load_ext sql")
+    ip_empty.run_cell(
+        """
+import duckdb
+conn = duckdb.connect()
+conn.execute('INSTALL httpfs')
```

Comment:
Do we need this?

Reply:
[Fixed](https://github.com/ploomber/jupysql/pull/912/files#diff-d327143929530bcece2cae65d59465708cdf87f499fd83b4bd939c332150f713R2066-R2069) - no need to install httpfs. I originally included that because the CI was returning an error when pulling data from https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv. No longer an issue since I switched to using our own test data